### PR TITLE
 🐛 Bug fix: Limit the Full-Width Responsive upgrade in Auto Ads to mobile users only.

### PR DESCRIPTION
**Pull request by Jiaming-X**
Code changes for this [issue](https://github.com/ampproject/amphtml/issues/28279)

### CODE REVIEW

File: extensions/amp-auto-ads/0.1/placement.js [hunk 1]
```diff
@@ -315,6 +319,18 @@ export class Placement {
...
+  /**
+   * Estimate if the viewport has a narrow layout.
+   * @param {!Element} element
+   * @return {boolean}
+   * @private
+   */
+  isLayoutViewportNarrow_(element) {
+    const viewportSize = Services.viewportForDoc(element).getSize();
+
+    return viewportSize.width < 488;
```
Comment (tlong2) [inline extensions/amp-auto-ads/0.1/placement.js:332]: Could you add a comment explaining the reasoning behind 488.

File: extensions/amp-auto-ads/0.1/test/test-placement.js [hunk 2]
```diff
@@ -608,16 +608,18 @@ describes.realWin(
...
-        env.sandbox.stub(mutator, 'requestChangeSize').callsFake(() => {
-          return Promise.resolve();
-        });
-        env.sandbox.stub(mutator.viewport_, 'getWidth').callsFake(() => 2000);
+        const viewportMock = env.sandbox.mock(
+          Services.viewportForDoc(env.win.document)
+        );
+        viewportMock
+          .expects('getSize')
+          .returns({width: 411, height: 823})
```
Comment (tlong2) [inline extensions/amp-auto-ads/0.1/test/test-placement.js:621]: It's good practice to test the boundary conditions. So in this scenario, we should have test cases testing width=487 and another for width=488, and then maybe a couple more test cases that test extreme values too just to check it still works as intended.

There is presumably a reason you chose 488 as the threshold, so you should have tests that will fail if someone inadvertently changes that value.

File: extensions/amp-auto-ads/0.1/placement.js (hunk 1 above)
Comment (Jiaming-X) [reply extensions/amp-auto-ads/0.1/placement.js:334]: Thanks. Added the comment. This is to align with our threshold for non-AMP traffics. You can code search isLayoutViewportNarrow in responsive_util.js for the internal reference.

File: extensions/amp-auto-ads/0.1/test/test-placement.js (hunk 2 above)
Comment (Jiaming-X) [reply extensions/amp-auto-ads/0.1/test/test-placement.js]: Thanks. Updated the width in the test case as you suggested.

**Pull request merged**